src/realm/query_expression.hpp

```diff
@@ -1217,15 +1222,31 @@ class Value : public ValueBase, public Subexpr2<T> {
     {
     }
 
-    std::string value_to_string(size_t ndx) const
+    std::string value_to_string(size_t ndx, util::serializer::SerialisationState& state) const
     {
         auto val = get(ndx);
         if (val.is_null())
             return "NULL";
         else {
+            static_cast<void>(state);
             if constexpr (std::is_same_v<T, TypeOfValue>) {
                 return util::serializer::print_value(val.get_type_of_value());
             }
+            else if constexpr (std::is_same_v<T, ObjKey>) {
+                ObjLink link(state.taget_table->get_key(), val.template get<ObjKey>());
```

Comment:
I'd prefer to have a defensive check in case the target table is null before dereferencing.

Reply:
OK